src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -839,16 +839,60 @@ int wm_vuldet_create_file(const char *path, const char *source) {
     const char *ROOT = "root";
     const char *sql;
     const char *tail;
-    sqlite3 *db = NULL;
-    sqlite3_stmt *stmt = NULL;
     int result;
     uid_t uid;
     gid_t gid;
+    bool already_removed = false;
+    bool db_existed = true;
+
+opendb: ;
+    sqlite3 *db = NULL;
+    sqlite3_stmt *stmt = NULL;
 
     // Open the database file, or create it
-    if (sqlite3_open_v2(path, &db, SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE, NULL)) {
-        mterror(WM_VULNDETECTOR_LOGTAG, VU_CREATE_DB_ERROR);
-        return wm_vuldet_sql_error(db, stmt);
+    if (sqlite3_open_v2(path, &db, SQLITE_OPEN_READWRITE, NULL)) {
+        db_existed = false;
+        if (sqlite3_open_v2(path, &db, SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE, NULL)) {
+            mterror(WM_VULNDETECTOR_LOGTAG, VU_CREATE_DB_ERROR);
+            return wm_vuldet_sql_error(db, stmt);
+        }
+    }
+
+    // Always try to create the metadata table
+    sqlite3_exec(db, vu_queries[CREATE_VU_METADATA], NULL, NULL, NULL);
+
+    if (db_existed && !already_removed) {
+        // Test db version, before load schema...
+        if (wm_vuldet_prepare(db, vu_queries[CHECK_DB_VERSION], -1, &stmt, NULL) != SQLITE_OK) {
+            return wm_vuldet_sql_error(db, stmt);
+        }
+        sqlite3_bind_text(stmt, 1, CVE_DB_VERSION, -1, NULL);
+
+        // ...delete and create a new one if schema version don't match
+        result = wm_vuldet_step(stmt);
+        if (result != SQLITE_ROW) {
+
+            wdb_finalize(stmt);
+            sqlite3_close_v2(db);
+            if (remove(path) < 0) {
+                mterror(WM_VULNDETECTOR_LOGTAG, DELETE_ERROR, path, errno, strerror(errno));
+                return OS_INVALID;
+            }
+
+            // Also will try to remove any journal file were created
+            const char db_suffix[] = "-journal";
+            char *db_journal;
+            int len = strlen(path) + strlen(db_suffix) + 1;
+
+            os_malloc(len, db_journal);
+            snprintf(db_journal, len, "%s%s", path, db_suffix);
+            (void)remove(db_journal);
+            os_free(db_journal);
+
+            mtdebug1(WM_VULNDETECTOR_LOGTAG, VU_DB_CHANGE_SCHEMA, path, CVE_DB_VERSION);
+            already_removed = true;
+            goto opendb;
```

Comment:
To improve the readability of the code, we will try to avoid the use of `goto` in this function. 
To do so, we have several alternatives, such as `if-else` structure, use of `while` or `do-while`, or _recursive function_.

Consider the different alternatives and try to choose the most efficient and readable one possible.